src/test/kotlin/com/pubnub/api/crypto/CryptoModuleTest.kt

```diff
@@ -0,0 +1,227 @@
+package com.pubnub.api.crypto
+
+import com.pubnub.api.crypto.cryptor.AesCbcCryptor
+import com.pubnub.api.crypto.cryptor.Cryptor
+import com.pubnub.api.crypto.cryptor.LegacyCryptor
+import com.pubnub.api.crypto.data.EncryptedData
+import com.pubnub.api.crypto.data.EncryptedStreamData
+import org.hamcrest.CoreMatchers
+import org.hamcrest.MatcherAssert
+import org.hamcrest.Matchers.hasSize
+import org.junit.jupiter.api.Assertions.assertArrayEquals
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import java.io.InputStream
+import org.hamcrest.Matchers.`is` as iz
+
+class CryptoModuleTest {
+
+    @Test
+    fun `can createLegacyCryptoModule`() {
+        // given
+        val cipherKey = "enigma"
+
+        // when
+        val legacyCryptoModule = CryptoModule.createLegacyCryptoModule(cipherKey)
+
+        // then
+        assertTrue(legacyCryptoModule.primaryCryptor is LegacyCryptor)
+        MatcherAssert.assertThat(legacyCryptoModule.cryptorsForDecryptionOnly, hasSize(2))
+        MatcherAssert.assertThat(
+            legacyCryptoModule.cryptorsForDecryptionOnly.first(),
+            iz(CoreMatchers.instanceOf(LegacyCryptor::class.java))
+        )
+        MatcherAssert.assertThat(
+            legacyCryptoModule.cryptorsForDecryptionOnly[1],
+            iz(CoreMatchers.instanceOf(AesCbcCryptor::class.java))
```

Comment:
Is the order of those really that important? We get them by id, so it shouldn't matter at all

Reply:
I see your point. I can modify this test to:

```
        assertThat(
            legacyCryptoModule.cryptorsForDecryptionOnly,
            containsInAnyOrder(
                listOf(
                    iz(CoreMatchers.instanceOf(AesCbcCryptor::class.java)),
                    iz(CoreMatchers.instanceOf(LegacyCryptor::class.java))
                )
            )
        )
```